test/app/core/managers/audio_record_services_test.dart

```diff
@@ -0,0 +1,313 @@
+    );
+  });
+
+  group(AudioRecordServices, () {
+    test(
+      'permissionStatus correct translate PermissionStatus to AudioPermissionState',
+      () async {
+        // arrange
+        final permissionMap = {
+          PermissionStatus.denied: AudioPermissionState.denied(),
+          PermissionStatus.granted: AudioPermissionState.granted(),
+          PermissionStatus.restricted: AudioPermissionState.restricted(),
+          PermissionStatus.permanentlyDenied:
+              AudioPermissionState.permanentlyDenied(),
+          PermissionStatus.limited: AudioPermissionState.undefined(),
+        };
+
+        permissionMap.forEach((key, value) async {
+          final mockPermissionHandlerPlatform =
+              PermissionHandlerPlatform.instance;
+          debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+          when((() => mockPermissionHandlerPlatform.checkPermissionStatus(
+              Permission.microphone))).thenAnswer((_) async => key);
+
+          // act
+          final result = await recordServices.permissionStatus();
+          // assert
+          expect(result, value,
+              reason: 'Expect value $value for permission status $key');
+        });
+      },
+    );
+
+    test(
+      'requestPermission return a AudioPermissionState.granted for PermissionStatus.granted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        // act
+        final result = await recordServices.requestPermission();
+
+        // assert
+        expect(result, AudioPermissionState.granted());
+      },
+    );
+
+    test(
+      'requestPermission return a AudioPermissionState.restricted for PermissionStatus.restricted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.restricted);
+
+        // act
+        final result = await recordServices.requestPermission();
+
+        // assert
+        expect(result, AudioPermissionState.restricted());
+      },
+    );
+
+    test(
+      'dispose() release the audio record session',
+      () async {
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => audioSyncManager.syncAudio())..thenAnswer((_) async => true);
+        when(() => soundRecorder.stopRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.closeRecorder()).thenAnswer((_) async => null);
```

Comment:
O retorno dos métodos `flutterSoundPlayer.stopRecorder()` e `flutterSoundPlayer.closeRecorder()` é `Future<void>`, nunca retornará nulo, acha que faria sentido o mock ser `void` também?

```suggestion
        when(() => soundRecorder.stopRecorder()).thenAnswer((_) async {});
        when(() => soundRecorder.closeRecorder()).thenAnswer((_) async {});
```